addons/product/product.py

```diff
@@ -432,6 +432,16 @@ class product_attribute_line(osv.osv):
         'value_ids': fields.many2many('product.attribute.value', id1='line_id', id2='val_id', string='Product Attribute Value'),
     }
 
+    def _check_valid_attribute(self, cr, uid, ids, context=None):
+        obj_pal = self.browse(cr, uid, ids[0], context=context)
+        if any(v in obj_pal.attribute_id.value_ids.ids for v in obj_pal.value_ids.ids):
```

Comment:
i tought about this and was thinking it was right

but you are right, if there is no `obj_pal.value_ids` this would return `False` wrongly if I understand correctly, so it should be flpped:

```python
        if any(v not in obj_pal.attribute_id.value_ids.ids for v in obj_pal.value_ids.ids):
            return False
        return True
```